kuwala/core/canvas/src/data/state/models/CanvasModel.js

```diff
@@ -119,6 +137,11 @@ const CanvasModel = {
 
             sourceDTO.connectedTargetNodeIds.push(target);
             updateTransformationBlock({ addNode, elements, setElements, updatedBlock: sourceDTO });
+        } else if (sourceElement.type === EXPORT_BLOCK) {
+            sourceDTO = sourceElement.data.exportBlock;
+
+            sourceDTO.connectedTargetNodeIds.push(target);
+            updateExportBlock({ addNode, elements, setElements, updatedBlock: sourceDTO });
```

Comment:
This action can be moved outside of all the blocks and only conditionally set the `sourceDTO`.